token/js/test/unit/transferHook.test.ts

```diff
@@ -1,173 +1,547 @@
+            expect(parsedExtraAccounts).to.not.be.null;
+            if (parsedExtraAccounts == null) {
+                return;
+            }
+
+            expect(parsedExtraAccounts).to.have.length(3);
+            if (parsedExtraAccounts.length !== 3) {
+                return;
+            }
+
+            expect(parsedExtraAccounts[0].discriminator).to.eql(0);
+            expect(parsedExtraAccounts[0].addressConfig).to.eql(plainAccount.toBuffer());
+            expect(parsedExtraAccounts[0].isSigner).to.be.false;
+            expect(parsedExtraAccounts[0].isWritable).to.be.false;
+
+            expect(parsedExtraAccounts[1].discriminator).to.eql(1);
+            expect(parsedExtraAccounts[1].addressConfig).to.eql(addressConfig);
+            expect(parsedExtraAccounts[1].isSigner).to.be.true;
+            expect(parsedExtraAccounts[1].isWritable).to.be.false;
+
+            expect(parsedExtraAccounts[2].discriminator).to.eql(128);
+            expect(parsedExtraAccounts[2].addressConfig).to.eql(addressConfig);
+            expect(parsedExtraAccounts[2].isSigner).to.be.false;
+            expect(parsedExtraAccounts[2].isWritable).to.be.true;
+        });
+
+        it('can resolve extra metas', async () => {
+            const resolvedPlainAccount = await resolveExtraAccountMeta(
+                connection,
+                plainExtraAccountMeta,
+                [],
+                instructionData,
+                testProgramId
+            );
+
+            expect(resolvedPlainAccount.pubkey).to.eql(plainAccount);
+            expect(resolvedPlainAccount.isSigner).to.be.false;
+            expect(resolvedPlainAccount.isWritable).to.be.false;
+
+            const resolvedPdaAccount = await resolveExtraAccountMeta(
+                connection,
+                pdaExtraAccountMeta,
+                [resolvedPlainAccount],
+                instructionData,
+                testProgramId
+            );
+
+            expect(resolvedPdaAccount.pubkey).to.eql(pdaPublicKey);
+            expect(resolvedPdaAccount.isSigner).to.be.true;
+            expect(resolvedPdaAccount.isWritable).to.be.false;
+
+            const resolvedPdaAccountWithProgramId = await resolveExtraAccountMeta(
+                connection,
+                pdaExtraAccountMetaWithProgramId,
+                [resolvedPlainAccount],
+                instructionData,
+                testProgramId
+            );
+
+            expect(resolvedPdaAccountWithProgramId.pubkey).to.eql(pdaPublicKeyWithProgramId);
+            expect(resolvedPdaAccountWithProgramId.isSigner).to.be.false;
+            expect(resolvedPdaAccountWithProgramId.isWritable).to.be.true;
         });
     });
 
-    it('getExtraAccountMetas', () => {
-        const accountInfo = {
-            data: extraAccountList,
-            owner: PublicKey.default,
-            executable: false,
-            lamports: 0,
+    // prettier-ignore
```

Comment:
Why is this needed?

Reply:
It nukes my bytes and comments below. I placed it up here when I had the tests with those large byte layouts, but now I can move it to the variable itself.